app/models/solidus_braintree/gateway.rb

```diff
@@ -364,6 +364,10 @@ def transaction_options(source, options, submit_for_settlement: false)
         params[:payment_method_nonce] = source.nonce
       end
 
+      if source.try(:device_data)
```

Comment:
```suggestion
      if source&.device_data
```

Nit. `&.` is pure Ruby and safer as it checks for the existence of the `device_data` method in case source is not already `nil`.